### ISSUE TITLE
BLE Mesh device lose when there are multiple BLE Mesh Gateway in local network

### ISSUE THREAD

**Issue by itispip** [AlexxIT/XiaomiGateway3#187]
Issues Replication:
1. Supposing you have only 1 BLE Mesh Gateway device (i.e. MiGateway3), and you are running AlexxIT's component v0.11.0. Initially you have 10 BLE Mesh device connected to gateway and HA will corrected show 10 devices as well
2. Now add a non-MiGateway3 BLE Mesh Gateway device (eg. Mi Smart Speaker Pro) into your network. After some days, some BLE Mesh devices will automatically roam to the Speaker depending on their distance to MiGateway3 & Speaker. You can see the roaming result in Mi SmartHome App.
3. Now, install a complete new instance of HA + AlexxIT's component, noticed it only creates device/entity for the BLE Mesh devices directly under the MiGateway3, it will not create device/entity for the BLE Mesh devices under Mi Smart Speaker. Even can still see all BLE MQTT msg as normal in MiGateWay3 MQTT server, but the component won't create them in HA.
4. Now turn-off Mi Smart Speaker, force the BLE Mesh Devices to roam back to MiGateway3. However AlexxIT's component still won't create device/entities for them.

**Comment by AlexxIT** [comment]
4. It's very strange because BLE devices can be added on the fly. What about MiHome? Can it see new devices?

**Comment by itispip** [comment]
> 1. It's very strange because BLE devices can be added on the fly. What about MiHome? Can it see new devices?

Hi Alex, MiHome will always display & control the BLE devices correctly, no matter the BLE device is connected into Mi via MiGateway3 or via other BLE Gateway such as Speaker or Camera. 

I was told BLE Devices are binded on MiAccount rather than on Gateway devices, so no matter which gateway the devices is using, MiHome will always be correct. And I also noticed that, no matter the devices is connected via which gateway, in MiGateWay3 MQTT server, the mqtt msg are always completely there, so these "lost" devices are actually still visible and controllable through MiGateWay3 if the component can properly discover these devices.

**Comment by itispip** [comment]
> > 1. It's very strange because BLE devices can be added on the fly. What about MiHome? Can it see new devices?
> 
> Hi Alex, MiHome will always display & control the BLE devices correctly, no matter the BLE device is connected into Mi via MiGateway3 or via other BLE Gateway such as Speaker or Camera.
> 
> I was told BLE Devices are binded on MiAccount rather than on Gateway devices, so no matter which gateway the devices is using, MiHome will always be correct. And I also noticed that, no matter the devices is connected via which gateway, in MiGateWay3 MQTT server, the mqtt msg are always completely there, so these "lost" devices are actually still visible and controllable through MiGateWay3 if the component can properly discover these devices.

Sorry to correct: the MQTT server does not have msg related to those "lost" BLE devices too. Mi SmartHome no problem

**Comment by itispip** [comment]
After I do below, the lost BLE devices (incluing BLE Light and Xiaomi BLE Sensor 2) comes back:
1. Delete the component from Integration
2. Delete the component from HACS
3. Reboot
4. Reinstall in HACS, but select version 0.10.4
5. Re-add into Integration

All the lost devices comes back. However, if I install 0.11.0 or Master, they won't appear.

@AlexxIT Something wrong with version 0.110 and after.

**Comment by relliky** [comment]
Hi @itispip, I have similar problem. I managed to re-discover these BLE devices without downgrading the integration by disabling BLE devices in the integraton options -> restart HA -> enabling BLE devices options -> restart HA.

I have two gateway.v3 and one of them died (becoming unable to stably connect devices). So I removed the died one and reconnect all devices to my secondary gateway.v3. However I don't see any BLE devices in HA. After I did the above steps, I managed to re-discover the BLE devices.
